c_glib/README.md

```diff
@@ -197,3 +197,37 @@ Now, you can run unit tests by the followings:
 % cd c_glib
 % test/run-test.sh
 ```
+
+## Common build problems
+### configure failed - `AX_CXX_COMPILE_STDCXX_11(ext, mandatory)'
+
+* Check whether `autoconf-archive` is installed.
+* [macOS] Run `brew install autoconf-archive` again. If it shows like the following message, run `brew link autoconf-archive`.
+
+```text
+$ brew install autoconf-archive
+Warning: autoconf-archive 2017.03.21 is already installed, it's just not linked.
+You can use `brew link autoconf-archive` to link this version.
+```
+
+Some packages (e.g. `gnome-common`) conflict with `autoconf-archive`. If you see like the following message, run `brew unlink <pkgname>`, then run `brew link autoconf-archive` again.
+
+```text
+$ brew link autoconf-archive
+Linking /usr/local/Cellar/autoconf-archive/2017.03.21... 
+Error: Could not symlink share/aclocal/ax_check_enable_debug.m4
+Target /usr/local/share/aclocal/ax_check_enable_debug.m4
+is a symlink belonging to gnome-common. You can unlink it:
+  brew unlink gnome-common
+```
+After installing/linking `autoconf-archive`, you need to run `./autogen.sh` again.
+
+### [macOS] configure failed - gobject-introspection-1.0 is not installed
+gobject-introspection requires libffi, and it's automatically installed with gobject-introspection. However it can't be found because it's [keg-only](https://docs.brew.sh/FAQ.html#what-does-keg-only-mean). You need to set `PKG_CONFIG_PATH` when executing configure.
+
+```text
+% PKG_CONFIG_PATH=$(brew --prefix libffi)/lib/pkgconfig ./configure
+```
+
+### build failed - /usr/bin/ld: cannot find -larrow
+Arrow C++ must be installed to build Arrow GLib. Run `make install` on Arrow C++ build directory. In addtion, on linux, you may need to run `sudo ldconfig`.
```

Comment:
Or we can use `--with-arrow-cpp-build-type` and `--with-arrow-cpp-build-dir` configure options.
It's better that description of them are included in this pull request but it's not required.

Reply:
#1100 and https://issues.apache.org/jira/browse/ARROW-1537 may be useful information for macOS.

Reply:
Thank you for useful advice. However I think such an advanced setting is not necessary for this "common build failures" section. It assumes that Arrow C++ is installed in accordance with cpp/README.md.